src/Sentry/Laravel/Features/Storage/Integration.php

```diff
@@ -23,15 +25,15 @@ public function isApplicable(): bool
 
     public function setup(): void
     {
-        $this->registerDiskDriver();
+        $this->registerDriver();
     }
 
     public function setupInactive(): void
     {
-        $this->registerDiskDriver();
+        $this->registerDriver();
     }
 
-    private function registerDiskDriver(): void
+    protected function registerDriver(): void
     {
         $this->container()->afterResolving(FilesystemManager::class, function (FilesystemManager $filesystemManager): void {
```

Comment:
Although this line of code was written by you in the previous [PR](https://github.com/getsentry/sentry-laravel/pull/726), I think it is not a good idea to extend the filesystemManager call when afterResolving it.  
This is because the FilesystemManager class may be resolved before the SentryServiceProvider, and therefore, the callback will never be called. 
```suggestion   
        $callback = function (FilesystemManager $filesystemManager): void {
            $filesystemManager->extend(self::STORAGE_DRIVER_NAME, function (Application $application, array $config) use ($filesystemManager): Filesystem {
                //...
            });
        };

        if ($this->container()->resolved(FilesystemManager::class)) {
            $callback($this->container()->get(FilesystemManager::class));
        } else {
            $this->container()->afterResolving(FilesystemManager::class, $callback);
        }
```

Reply:
That should never happen. We register in the `register` callback of a service provider. And resolving should have never happened there so this should be perfectly fine unless somethin wonky is going on. Do you have a specific use case in mind where this would actually fail?

Reply:
Yes, it actually happend in our project, but I need to time to reproduce in a simple Laravel project.

Reply:
I guess you have a service provider that is not well behaved, meaning that it does more than just register bindings.

Reply:
Or you register the Sentry service provider manually and from inside another service provider, but we use the `afterResolving` callback on more places and I've never seen the issues you describe or heard people having them.

Reply:
Indeed, we have a service provider that calls `$this->app->make('filesystem')` in the `boot` method.

Reply:
Ah right, we are doing this in the `boot` not `register`. I'll see what we can do to change that. But your suggestion wouldn't currently fix the problem since it will still be too late.

Reply:
@yankewei it took a bit, but I think we solved the issue and now register the driver before Laravel is booted which should solve the issues you were seeing!